Toggle Hide/Show search form #138068115

patmbolger opened the pull request:
PT Story: https://www.pivotaltracker.com/story/show/138068115


Changes proposed in this pull request:
1. Add control to companies index view to hide / show search form
2. Add Gem ```i18n-js``` for i18n functionality (via Rails config) in client
3. Add JS method to toggle visibility of an element
4. Add embedded script to layout view to set i18n default and current locale in client (when  user changes locale / language)
5. Add feature test for toggle 

Screenshots (Optional):
<hr>
<img width="733" alt="screen shot 2017-01-29 at 4 38 41 pm" src="https://cloud.githubusercontent.com/assets/9968213/22408138/8f9800c2-e641-11e6-89ef-b89289888364.png">

<hr>
<img width="776" alt="screen shot 2017-01-29 at 4 39 02 pm" src="https://cloud.githubusercontent.com/assets/9968213/22408141/960c9008-e641-11e6-8ad3-ebd18a15e085.png">

<hr>
<img width="720" alt="screen shot 2017-01-29 at 4 39 13 pm" src="https://cloud.githubusercontent.com/assets/9968213/22408143/9aaff0a0-e641-11e6-81e8-f01075c7580a.png">

<hr>
<img width="719" alt="screen shot 2017-01-29 at 4 39 24 pm" src="https://cloud.githubusercontent.com/assets/9968213/22408144/9f1c4b5c-e641-11e6-9b25-292557889ef4.png">



Ready for review:
@thesuss @weedySeaDragon

patmbolger committed:
interim progress

patmbolger committed:
hide/show company search form

patmbolger committed:
fixed cuke tests

thesuss commented on config/locales/sv.yml:
```diff
@@ -43,6 +43,10 @@ sv:
   org_nr: &org_nr Org no.
 
+  toggle:
+    hide: Dölja
```
Dölj 

(Dölja = to hide)

thesuss commented:
```
    Then I wait for 1 second                                                              # features/step_definitions/basic_steps.rb:103
    Then I select "Groomer" in select list t("activerecord.models.business_category.one") # features/step_definitions/basic_steps.rb:98
      Unable to find select box "Kategori" (Capybara::ElementNotFound)
      ./features/step_definitions/basic_steps.rb:100:in `/^(?:I|they) select "([^"]*)" in select list t\("([^"]*)"\)$/'
      features/search_companies.feature:53:in `Then I select "Groomer" in select list t("activerecord.models.business_category.one")'
```

Maybe wait longer?

patmbolger committed:
fixed translation, increase XHR wait time

patmbolger committed:
increased wait times again

diraulo reviewed (approved):
The build is all good now. Sorry but didn't get a chance to look at this earlier. All that was missing was the setting @patmbolger suggested and @thesuss has access to Semaphore so you could've made that change too 😉 

We should probably move this project to the AV org at some point, that'll give members of the teams access to modify the build settings as necessary in future.

Also, you might want to consider an alternative way of including external dependencies (JS libraries, etc.. ) they're hurting your GPA on codeclimate so badly.

patmbolger committed:
houndci issues fix

diraulo commented on Gemfile:
```diff
@@ -23,6 +23,7 @@ gem 'routing-filter'   # for handling locale filters around routes
 
 gem 'ransack'
 gem 'city-state'
+gem 'i18n-js', '>= 3.0.0.rc11'
```
👍

patmbolger replied:
@diraulo - thanks, I'll try that.

diraulo reviewed (commented):
@patmbolger now that you have the `i18n-js` gem in place do you still need `public/javascripts/i18n.js`?

I think you can just add `= require i18n.js` and remove the file in the js folder.

patmbolger committed:
added jshint ignore file

patmbolger committed:
hound, raoul suggestions

patmbolger committed:
cut wait time to 2 seconds cmpy search feature test

weedySeaDragon commented on config/locales/en.yml:
```diff
@@ -281,7 +285,7 @@ en:
       region_land: Region
       h_companies_listed_below: Member companies are listed below.
-      how_to_search: Search by specifying one or more of the search parameters below.
+      how_to_search: Search for companies in search form.
```
grammar fix: need `the`

`how_to_search: Search for companies in the search form.`

or something a little more explicit (_using..._):

`Search for companies using the search form.`

weedySeaDragon reviewed (changes requested):
Looking good. (and all tests pass on my machine.  hooray!)

1 change that needs to be made to en.yml, and these UX suggestions (which could be done in a different PR):

1. I don't think `(Show)` and `(Hide)` are clear enough.  I don't think they communicate that they are connected to the search form.  I'd rather see something like:

    `Search for companies using the search form. (Show the search form.)`

   and 

    `...     (Hide the search form.)`

    And include an icon to for the 'show/expand' and 'hide/collapse' idea as well. Maybe a down triangle and up triangle.



2. The search form disappears when the results are displayed.  We need some visual indicator/cue that what the user is seeing is the result of a search with some criteria/filters.  Either don't hide the search form or show the criteria somehow.

patmbolger committed:
changes after PR review

patmbolger commented:
Thanks for the review @weedySeaDragon 

I changed the wording on the search form along the lines of your suggestions.

Regarding your second comment, I added this chore: https://www.pivotaltracker.com/story/show/139001697

thesuss commented on config/locales/sv.yml:
```diff
@@ -43,6 +43,11 @@ sv:
 
+  toggle:
+    company_search_form:
+      hide: Dölja sökfunktionen
```
Dölj sökfunktionen 
(no a)

thesuss commented on config/locales/sv.yml:
```diff
@@ -279,7 +284,7 @@ sv:
       region_land: *region_short
       h_companies_listed_below: Nedan listas anslutna företag.
-      how_to_search: Sök genom att ange en eller flera av sökparametrarna nedan.
+      how_to_search: Sök efter företag som använder sökfunktionen.
```
Sök efter företag genom att använda sökfunktionen.

patmbolger committed:
corrected translations

thesuss committed:
Merge branch 'develop' into toggle-search#138068115
